### PR TITLE
[SDL-0273] Widgets processing for WEB_VIEW template

### PR DESCRIPTION
**Pull request by VjKlepikov**
https://adc.luxoft.com/jira/browse/FORDTCN-6816

ATF Test Scripts to check the creation of widget when application has active WEB_VIEW template

This PR is **[ready]** for review.

### Summary

- Processing the creation of widget which do not duplicate main window when application has active WEB_VIEW template
- Processing the creation of widget which duplicate main window
- Processing of RPCs for widgets which do not duplicate main window
- Processing of RPCs for widget which duplicate main window

### ATF version
[develop]

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/WebEngine/WebEngineProjection/014_Show_RPC_to_Main_Window_widget_with_DuplicateID.lua [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0273-webengine-projection-mode.md
+--
+-- Description: Processing of RPCs for widget which duplicate main window ( "duplicateUpdatesFromWindowID" = 0 )
+--  when application has active WEB_VIEW template
+--
+-- Preconditions:
+-- 1) SDL and HMI are started
+-- 2) WebEngine App with WEB_VIEW HMI type is registered
+-- 3) App successfully created a widget with "duplicateUpdatesFromWindowID" parameter for Main window
+-- 4) Widget is activated on the HMI and has FULL level
```
Comment (LitvinenkoIra) [inline test_scripts/WebEngine/WebEngineProjection/014_Show_RPC_to_Main_Window_widget_with_DuplicateID.lua:11]: @VjKlepikov Please specify that you have two widgets

File: test_scripts/WebEngine/commonWebEngine.lua [hunk 2]
```diff
@@ -237,6 +237,65 @@ function common.activateWidgetFromNoneToFULL(pId, pAppId)
...
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  common.getMobileSession(pAppId):ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+  :Do(function()
+      if params.requestShowParams.templateConfiguration ~= nil then
+        local paramsToSDL = common.getOnSystemCapabilityParams()
+        paramsToSDL.appID = common.getHMIAppId(pAppId)
+        common.getHMIConnection():SendNotification("BasicCommunication.OnSystemCapabilityUpdated", paramsToSDL)
+        common.getMobileSession(pAppId):ExpectNotification("OnSystemCapabilityUpdated",
+          common.getOnSystemCapabilityParams())
```
Comment (LitvinenkoIra) [inline test_scripts/WebEngine/commonWebEngine.lua:291]: @VjKlepikov I think that _getOnSystemCapabilityParams_ function should be updated... Here https://github.com/LuxoftSDL/sdl_atf_test_scripts/blob/1016b4a3ffbf90cf4aefdfa1edb3a51ce6ed5679/test_scripts/WebEngine/commonWebEngine.lua#L516 we should specify WEB_VIEW template as available. Could you please check it

Comment (VjKlepikov) [reply test_scripts/WebEngine/commonWebEngine.lua:291]: @LitvinenkoIra getOnSystemCapabilityParams contains update for  widget (windowID = 1). According to propousal
the WEB_VIEW template is available only to the main window. It should not be available for widgets. So we shouldn't specify WEB_VIEW template as available for the widget (windowID = 1)

File: test_scripts/WebEngine/WebEngineProjection/013_Show_RPC_to_Main_Window_widget_without_DuplicateID.lua [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+local createWindowParams = {
+  windowID = 1,
+  windowName = "Name",
+  type = "WIDGET"
+}
+
+--[[ General configuration parameters ]]
+config.application1.registerAppInterfaceParams.appHMIType = appHMIType
+config.application1.registerAppInterfaceParams.syncMsgVersion.majorVersion = 7
+config.application1.registerAppInterfaceParams.syncMsgVersion.minorVersion = 0
```
Comment (YarikMamykin) [inline test_scripts/WebEngine/WebEngineProjection/013_Show_RPC_to_Main_Window_widget_without_DuplicateID.lua:40]: I would propose to backmerge current feature branch and use `common.setupRAIParams` function for this.
This relates to other test scripts.

**Comment by VjKlepikov** [comment]
@aderiabin thank you for review. Please check in [c20607e](https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/33/commits/c20607e79de1949a79929d853f251aba9b9114fb)

File: test_scripts/WebEngine/WebEngineProjection/013_Show_RPC_to_Main_Window_widget_without_DuplicateID.lua (hunk 3 above)
Comment (VjKlepikov) [reply test_scripts/WebEngine/WebEngineProjection/013_Show_RPC_to_Main_Window_widget_without_DuplicateID.lua]: Done in [(173089)](https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/33/commits/173089eb6bf1a528d8f509b7f7f2916ee1b1e1d8)

**Pull request merged**